### ISSUE TITLE
Actions workflow for 3.4 passed despite failures within tests

### ISSUE THREAD

**Issue by jmhbnz** [etcd-io/etcd#15487]
### Which github workflows are flaking?

This workflow run succeeded despite me having errors with one of the tests defined: https://github.com/etcd-io/etcd/actions/runs/4433391180 

Issue was with file `tests/e2e/etcd_config_test.go`

### Which tests are flaking?

Specifically this job: https://github.com/etcd-io/etcd/actions/runs/4433391180/jobs/7778376779

Which finished with the following job output:
```bash
# go.etcd.io/etcd/tests/e2e [go.etcd.io/etcd/tests/e2e.test]
Error: tests/e2e/etcd_config_test.go:342:2: undefined: skipInShortMode
Error: tests/e2e/etcd_config_test.go:362:6: too many arguments in call to spawnCmd
	have ([]string, nil)
	want ([]string)
FAIL	go.etcd.io/etcd/tests/e2e [build failed]
FAIL
```

### Github Action link

https://github.com/etcd-io/etcd/actions/runs/4433391180/jobs/7778376779

### Anything else we need to know?

Expectation is this kind of failure would cause the job to fail.

Relates to https://github.com/etcd-io/etcd/pull/15486

**Comment by fuweid** [comment]
/assign 

I'm debugging some flaky testcases in pipeline. Let me handle this. Thanks

**Comment by fuweid** [comment]
Closed by #15504
